pkg/karmadactl/factory.go

```diff
@@ -0,0 +1,104 @@
+package karmadactl
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+
+	karmadaclientset "github.com/karmada-io/karmada/pkg/generated/clientset/versioned"
+)
+
+const proxyURL = "/apis/cluster.karmada.io/v1alpha1/clusters/%s/proxy/"
+
+// The FactoryExpansion interface provides 2 major features compared to the cmdutil.Factory:
```

Comment:
Could it support to create `Factory` for `karmada control plane`? If we need this for `karmada control plane` in future, I think we only need to call our own `util.Factory`.